app/components/ExamPages/ScalesNotateTemplate.tsx

```diff
@@ -14,9 +19,17 @@ export default function ScalesNotation({
   nextViewState,
   page,
 }: UserDataProps) {
+  const [scaleDataMatrix, setScaleDataMatrix] = useState<ScaleData[][]>(
+    currentUserData[`scaleDataMatrix${page - 5}`] || [[]]
+  );
+  const [scaleStaves, setScaleStaves] = useState<StaveType[]>(
+    currentUserData[`scaleStaves${page - 5}`] || []
+  );
   const [scales, setScales] = useState<Array<string>>([]);
 
   const scalesPropName = `scales${page - 5}`;
+  const scaleDataMatrixPropName = `scaleDataMatrix${page - 5}`;
+  const stavesPropName = `scaleStaves${page - 5}`;
 
   const handleSubmit = async (e: FormEvent) => {
     e.preventDefault();
```

Comment:
Are you missing scaleDataMatrixPropName, stavesPropName from the handleSubmit function? Even though this hasn't been implemented, it would make it more consistent like in KeySigNotateTemplate